src/coreclr/tools/aot/ILCompiler/RdXmlRootProvider.cs

```diff
@@ -160,7 +160,7 @@ static bool SignatureMatches(MethodDesc method, List<TypeDesc> parameter)
 
                 for (int i = 0; i < method.Signature.Length; i++)
                 {
-                    if (method.Signature[i] != parameter[i])
+                    if (!method.Signature[i].IsSignatureVariable && method.Signature[i] != parameter[i])
```

Comment:
```suggestion
                    if (!method.Signature[i].ContainsSignatureVariables() && method.Signature[i] != parameter[i])
```

This will also catch things like `void Method(List<T> parameter)` as opposed to just `void Method(T parameter)`.

Reply:
I've update the code to match it with instArgs.

Reply:
Done in 9410939